aws/resource_aws_elasticsearch_domain.go

```diff
@@ -443,6 +462,26 @@ func resourceAwsElasticSearchDomainCreate(d *schema.ResourceData, meta interface
 		}
 	}
 
+	if v, ok := d.GetOk("domain_endpoint_options"); ok {
+		options := v.([]interface{})
+
+		if len(options) == 1 {
+			if options[0] == nil {
+				return fmt.Errorf("At least one field is expected inside domain_endpoint_options")
+			}
+
+			s := options[0].(map[string]interface{})
+			domainOptions := elasticsearch.DomainEndpointOptions{}
+			if requireHTTPS, ok := s["require_https"]; ok {
+				domainOptions.EnforceHTTPS = aws.Bool(requireHTTPS.(bool))
+			}
+			if tls, ok := s["tls_security_policy"]; ok {
+				domainOptions.TLSSecurityPolicy = aws.String(tls.(string))
+			}
+			input.SetDomainEndpointOptions(&domainOptions)
+		}
+	}
```

Comment:
This type of logic can usually be simplified by creating an "expand" function (similar to `expandESCognitoOptions` below): 👍 

```go
	if v, ok := d.GetOk("domain_endpoint_options"); ok {
		input.DomainEndpointOptions = expandESDomainEndpointOptions(v.([]interface{}))
	}

// ...

func expandESDomainEndpointOptions(l []interface{}) *elasticsearch.DomainEndpointOptions {
	if len(l) == 0 || l[0] == nil {
		return nil
	}

	m := l[0].(map[string]interface{})
	domainEndpointOptions := &elasticsearch.DomainEndpointOptions{}

	if v, ok := m["enforce_https"].(bool); ok {
		domainEndpointOptions.EnforceHTTPS = aws.Bool(v)
	}

	if v, ok := m["tls_security_policy"].(string); ok {
		domainEndpointOptions.TLSSecurityPolicy = aws.String(v)
	}

	return domainEndpointOptions
}
```